lib/artifactory/resources/artifact.rb

```diff
@@ -620,6 +631,45 @@ def upload_from_archive(repo, remote_path, properties = {})
     private
 
     #
+    # Helper method for reading artifact properties
+    #
+    # @example List all properties for an artifact
+    #   artifact.get_properties #=> { 'artifactory.licenses'=>['Apache-2.0'] }
+    #
+    # @param [TrueClass, FalseClass] refresh_cache (default: +false+)
+    #   wether or not to use the locally cached value if it exists and is not nil
+    #
+    # @return [Hash<String, Object>]
+    #   the list of properties
+    #
+    def get_properties(refresh_cache = false)
+      if refresh_cache || @properties.nil?
+        @properties = client.get(File.join("/api/storage", relative_path), properties: nil)["properties"]
+      end
+
+      @properties
+    end
+
+    #
+    # Helper method for setting artifact properties
+    #
+    # @example Set properties for an artifact
+    #   artifact.set_properties({ prop1: 'value1', 'prop2' => 'value2' })
+    #
+    # @param [Hash<String, Object>] props
+    #   A hash of properties and corresponding values to set for the artifact
+    #
+    # @return [Hash]
+    #   the parsed JSON response from the server
+    #
+    def set_properties(props)
+      safe_properties = props.map { |p, v| "#{url_safe(p)}=#{url_safe(v)}" }.join(";")
+      artifact_url_with_props = "#{File.join("/api/storage", relative_path)}?properties=#{safe_properties}"
```

Comment:
We can simplify this by taking advantage of the `Artifactory::Resource::Base#to_matrix_properties ` helper:

```ruby
matrix   = to_matrix_properties(properties)
endpoint = File.join("/api/storage", relative_path) + "?properties=#{matrix}"

client.put(endpoint, nil)
```